polkadot/xcm/pallet-xcm/src/tests.rs

```diff
@@ -433,186 +560,1570 @@ fn limited_teleport_assets_works() {
+		assert_eq!(Assets::total_issuance(usdt_id_multilocation), expected_usdt_issuance);
+		assert_eq!(Assets::active_issuance(usdt_id_multilocation), expected_usdt_issuance);
+		// Verify total and active issuance of foreign BLA asset have decreased (burned on
+		// reserve-withdraw)
+		let expected_bla_issuance = foreign_initial_amount - SEND_AMOUNT;
+		assert_eq!(Assets::total_issuance(foreign_asset_id_multilocation), expected_bla_issuance);
+		assert_eq!(Assets::active_issuance(foreign_asset_id_multilocation), expected_bla_issuance);
+
+		// Verify sent XCM program
+		assert_eq!(
+			sent_xcm(),
+			vec![
+				(
+					// first message is to prefund fees on `dest`
+					dest,
+					// fees are teleported to destination chain
+					Xcm(vec![
+						ReceiveTeleportedAsset(expected_fee.clone().into()),
+						ClearOrigin,
+						buy_limited_execution(expected_fee.clone(), Unlimited),
+						DepositAsset { assets: AllCounted(1).into(), beneficiary },
+					])
+				),
+				(
+					// second message is to transfer/deposit foreign assets on `dest` while paying
+					// using prefunded (transferred above) fees (USDT)
+					// (dest is reserve location for `expected_asset`)
+					dest,
+					Xcm(vec![
+						WithdrawAsset(expected_asset.into()),
+						ClearOrigin,
+						buy_limited_execution(expected_fee, Unlimited),
+						DepositAsset { assets: AllCounted(1).into(), beneficiary },
+					])
+				)
+			]
+		);
+		let versioned_sent = VersionedXcm::from(sent_xcm().into_iter().next().unwrap().1);
+		let _check_v2_ok: xcm::v2::Xcm<()> = versioned_sent.try_into().unwrap();
+	});
+}
+
+/// Test `reserve_transfer_assets` with remote asset reserve and teleported fee.
+///
+/// Transferring foreign asset (reserve on `FOREIGN_ASSET_RESERVE_PARA_ID`) to `USDT_PARA_ID`. Using
+/// teleport-trusted USDT for fees.
+///
+/// ```nocompile
+///    | chain `A`       |  chain `C`                      |  chain `B`
+///    | Here (source)   |  FOREIGN_ASSET_RESERVE_PARA_ID  |  USDT_PARA_ID (destination)
+///    |                 |  `assets` reserve               |  `fees` (USDT) teleport-trust
+///    |
+///    |  1. `A` executes `InitiateTeleport(fees)` dest `C`
+///    |     \---------->  `C` executes `ReceiveTeleportedAsset(fees), .., DepositAsset(fees)`
+///    |
+///    |  2. `A` executes `InitiateTeleport(fees)` dest `B`
+///    |     \------------------------------------------------->  `B` executes:
+///    |                                `ReceiveTeleportedAsset(fees), .., DepositAsset(fees)`
+///    |
+///    |  3. `A` executes `InitiateReserveWithdraw(assets)` dest `C`
+///    |     -----------------> `C` executes `DepositReserveAsset(assets)` dest `B`
+///    |                             --------------------------> `DepositAsset(assets)`
+///    |  all of which at step 3. being paid with fees prefunded in steps 1 & 2
+/// ```
+#[test]
+fn reserve_transfer_assets_with_remote_asset_reserve_and_teleported_fee_works() {
```

Comment:
@acatangiu 
just thinking out loud, what about moving these tests to the separate file like `tests_for_asset_transfer.rs`?

Reply:
not a fan of multiple test files, but not against it either if this proposal gains traction

Reply:
I am not a fan of 2000+ lines files, so I support the proposal 😄 
We can make a `tests` folder and put the different test files in there